how-to/containerd-kata.md

```diff
@@ -0,0 +1,357 @@
+    [plugins.cri.containerd]
+      [plugins.cri.containerd.default_runtime]
+        #runtime_type = "io.containerd.runtime.v1.linux"
+
+    [plugins.cri.cni]
+      # conf_dir is the directory in which the admin places a CNI conf.
+      conf_dir = "/etc/cni/net.d"
+```
+
+Let's add Kata Containers to the configurations.
+
+#### Kata Containers as a RuntimeClass
+
+For 
+- Kata Containers v1.5.0 or above (including 1.5.0-rc)
+- Containerd v1.2.0 or above
+- Kubernetes v1.12.0 or above
+
+The  RuntimeClass is suggested.
+
+The following configuration include three runtime classes:
+- `plugins.cri.containerd.runtimes.runc`: the runC, and it is the default runtime;
+- `plugins.cri.containerd.runtimes.kata`: it is magic in containerd (ref [the doc here](https://github.com/containerd/containerd/tree/master/runtime/v2#binary-naming)), in which the dot connected string `io.containerd.kata.v2` will be translated to `containerd-shim-kata-v2`, i.e. the binary name of the kata implementation of shim v2 API;
+[Containerd Runtime V2 (Shim API)](https://github.com/containerd/containerd/tree/master/runtime/v2), the 
+- `plugins.cri.containerd.runtimes.katacli`: the `containerd-shim-runc-v1` will call `kata-runtime`, which is the legacy way deprecated by the `containerd-shim-kata-v2`:
+
+```toml
+    [plugins.cri.containerd]
+      no_pivot = false
+    [plugins.cri.containerd.runtimes]
+      [plugins.cri.containerd.runtimes.runc]
+         runtime_type = "io.containerd.runc.v1"
+         [plugins.cri.containerd.runtimes.runc.options]
+           NoPivotRoot = false
+           NoNewKeyring = false
+           ShimCgroup = ""
+           IoUid = 0
+           IoGid = 0
+           BinaryName = "runc"
+           Root = ""
+           CriuPath = ""
+           SystemdCgroup = false
+      [plugins.cri.containerd.runtimes.kata]
+         runtime_type = "io.containerd.kata.v2"
+      [plugins.cri.containerd.runtimes.katacli]
+         runtime_type = "io.containerd.runc.v1"
+         [plugins.cri.containerd.runtimes.katacli.options]
+           NoPivotRoot = false
+           NoNewKeyring = false
+           ShimCgroup = ""
+           IoUid = 0
+           IoGid = 0
+           BinaryName = "/usr/bin/kata-runtime"
+           Root = ""
+           CriuPath = ""
+           SystemdCgroup = false
+```
+
+Based on the discussion in [containerd/containerd#2916](https://github.com/containerd/containerd/pull/2916),
+the containerd configuration will have a field for runtime configuration files (likely to be`config_file`). However, 
+the field has not been available yet and the `shimv2` will search configuration file in the default paths 
+(same as `kata-runtime` cli): `/etc/kata-containers/configuration.toml`, `/usr/share/defaults/kata-containers/configuration.toml`. 
+And the document will be updated once the `config_file` field is available.
+
+If you want to specify a configure file now, here is a workaround. The `shimv2` accept an environment variable,
```

Comment:
Lines 193-194 suggested rewrite: 
If you want to specify a configure file now, the following provides a workaround. The `shimv2` accepts an environment variable, `KATA_CONF_FILE` for the config file path. Then, you can create a shell script with the following: